websockify/websocketproxy.py

```diff
@@ -257,19 +260,24 @@ def do_proxy(self, target):
                 # Receive target data, encode it and queue for client
                 buf = target.recv(self.buffer_size)
                 if len(buf) == 0:
-
+                    # Target socket closed, flushing queues and closing client-side websocket
                     # Send queued target data to the client
-                    c_pend = self.send_frames(cqueue)
-                    cqueue = []
+                    while (len(cqueue) != 0):
+                        # As do-until loops are not available in python, settings c_pend to true
```

Comment:
I don't think we need programming language commentary. :)

You should also be able to do:

```python3
while (self.send_frames(cqueue)):
    pass
```